src/test/fixtures/mocks/window-history.js

```diff
@@ -0,0 +1,67 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import { coerceMatchingShape } from '../../../utils/flow';
+
+/**
+ * jsdom leaves the history in place after every test, so the history will be dirty.
+ * Plus when we stop using a component it could still be subscribed to events, which
+ * does not really reflect real-world use cases. This mock creates a history API
```

Comment:
I'm not so sure of this sentence: if a component doesn't unsubscribe from events, I think that would be a bug. But it's not a big deal for _these_ events in our project because our components doing this are long-living.

Reply:
Ok, I updated the comment, and make sure we didn't have anything leaky. I don't think my comment was accurate anyway, as it was on the `window` object.

```diff
diff --git a/src/components/app/UrlManager.js b/src/components/app/UrlManager.js
index 38be5735..23022459 100644
--- a/src/components/app/UrlManager.js
+++ b/src/components/app/UrlManager.js
@@ -124,7 +124,7 @@ class UrlManager extends React.PureComponent<Props> {
     }
   }
 
-  _updateState() {
+  _updateState = () => {
     const { updateUrlState, show404, urlState: previousUrlState } = this.props;
     let newUrlState;
     if (window.history.state) {
@@ -161,11 +161,15 @@ class UrlManager extends React.PureComponent<Props> {
 
     // Update the Redux store.
     updateUrlState(newUrlState);
-  }
+  };
 
   componentDidMount() {
     this._processInitialUrls();
-    window.addEventListener('popstate', () => this._updateState());
+    window.addEventListener('popstate', this._updateState);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('popstate', () => this._updateState);
   }
 
   UNSAFE_componentWillReceiveProps(nextProps: Props) {
diff --git a/src/test/fixtures/mocks/window-history.js b/src/test/fixtures/mocks/window-history.js
index d948de18..3a6796d8 100644
--- a/src/test/fixtures/mocks/window-history.js
+++ b/src/test/fixtures/mocks/window-history.js
@@ -7,9 +7,7 @@ import { coerceMatchingShape } from '../../../utils/flow';
 
 /**
  * jsdom leaves the history in place after every test, so the history will be dirty.
- * Plus when we stop using a component it could still be subscribed to events, which
- * does not really reflect real-world use cases. This mock creates a history API
- * that can be thrown away after every use.
+ * This mock creates a history API that can be thrown away after every use.
  */
 export function mockWindowHistory() {
   const originalHistory = Object.getOwnPropertyDescriptor(window, 'history');
```